stubs/numba.pyi

```diff
@@ -0,0 +1,41 @@
+from typing import (
+    Any,
+    Callable,
+    Literal,
+    overload,
+)
+
+import numba
+
+from pandas._typing import F
+
+def __getattr__(name: str) -> Any: ...  # incomplete
+@overload
+def jit(
+    signature_or_function: F = ...,
+) -> F: ...
+@overload
+def jit(
+    signature_or_function: str
+    | list[str]
+    | numba.core.types.abstract.Type
+    | list[numba.core.types.abstract.Type] = ...,
+    locals: dict = ...,  # TODO: Mapping of local variable names to Numba types
+    cache: bool = ...,
+    pipeline_class: numba.compiler.CompilerBase = ...,
+    boundscheck: bool | None = ...,
+    *,
+    nopython: bool = ...,
+    forceobj: bool = ...,
+    looplift: bool = ...,
+    error_model: Literal["python", "numpy"] = ...,
+    inline: Literal["never", "always"] | Callable = ...,
+    # TODO: If a callable is provided it will be called with the call expression
+    # node that is requesting inlining, the caller's IR and callee's IR as
+    # arguments, it is expected to return Truthy as to whether to inline.
+    target: Literal["cpu", "gpu", "npyufunc", "cuda"] = ...,  # deprecated
```

Comment:
we could leave this out, we don't currently use this and since it is deprecated may want to not do so.

Reply:
Generally we only use `signature_or_function`, `nopython`, `nogil`, `parallel` if that would simplify the typing. `looplift` may be utilized in the future.

Reply:
There's an open issue on numba tracker https://github.com/numba/numba/issues/7424

The open PR to fix only deals with the decorator preserving types

I will also be using this stub in my numba branch and adding additional types as needed. Will look at the types for the numba types next if this PR is accepted.

It maybe that being more complete helps the community to share effort until official numba stubs are available.